From 9737316400eac39364224b4691c75d632314bb0b Mon Sep 17 00:00:00 2001
From: John Wason <wason@wasontech.com>
Date: Fri, 17 Apr 2026 21:34:25 -0400
Subject: [PATCH] Fix eigen 5 version in cmake package config files

---
 cmake/Config.cmake.in | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/cmake/Config.cmake.in b/cmake/Config.cmake.in
index b4c8633..c706756 100644
--- a/cmake/Config.cmake.in
+++ b/cmake/Config.cmake.in
@@ -9,4 +9,7 @@ set(RobotRaconteur_STANDARD_ROBDEF_FILES @RR_INSTALLED_ROBDEF@)
 set(RobotRaconteur_STANDARD_ROBDEF_DIRS @RR_INSTALLED_ROBDEF_DIRS@)
 
 find_package(yaml-cpp REQUIRED)
-find_package(Eigen3 3.3 REQUIRED NO_MODULE)
+find_package(Eigen3 3.3 NO_MODULE QUIET)
+if(NOT Eigen3_FOUND)
+    find_package(Eigen3 3.3...<6 REQUIRED NO_MODULE)
+endif()
